lib/Server.js

```diff
@@ -2594,14 +2594,18 @@ class Server {
    * @returns {void}
    */
   runBonjour() {
+    const { Bonjour } = require("bonjour-service");
     /**
      * @private
-     * @type {import("bonjour").Bonjour | undefined}
+     * @type {Bonjour | undefined}
      */
-    this.bonjour = require("bonjour")();
+    this.bonjour = new Bonjour();
     this.bonjour.publish({
+      // @ts-ignore
       name: `Webpack Dev Server ${os.hostname()}:${this.options.port}`,
+      // @ts-ignore
       port: /** @type {number} */ (this.options.port),
+      // @ts-ignore
```

Comment:
Why ignore?

Reply:
<img width="719" alt="Screenshot 2022-02-24 at 6 53 45 AM" src="https://user-images.githubusercontent.com/46647141/155439427-943741e4-c942-4175-b136-87cba30ecee3.png">

Otherwise, we have the above error because of the `spread` below. There seems to be an open issue in TypeScript about this - https://github.com/microsoft/TypeScript/issues/37559

Reply:
Maybe we need `// @ts-expect-error` here.

Reply:
Yep, `// @ts-expect-error` is logically here